src/components/App/App.vue

```diff
@@ -43,9 +49,46 @@ export default {
       })
     },
     retrieveLocale (location) {
-      fetchAPI.getLocale(location.city, location.state).then(data => {
-        this.locations.push(data.data)
+      fetchAPI.getLocale(location.city, location.state).then(value => {
+        value.data.id = Date.now()
+        this.checkExistingLocations(value.data)
+        this.updateLocalStorage()
+      })
+    },
+    retrieveCurrent () {
+      fetchAPI.getCurrent().then(location => {
+        location.data.id = Date.now()
+        this.checkExistingLocations(location.data)
+        this.updateLocalStorage()
+      })
+    },
+    removeLocation (id) {
+      const thisLocation = this.locations.find(location => {
+        return location.id === parseInt(id)
       })
+      const thisLocationIndex = this.locations.indexOf(thisLocation)
+      this.locations.splice(thisLocationIndex, 1)
+      this.updateLocalStorage()
+    },
+    checkExistingLocations (newLocation) {
+      const foundLocation = this.locations.find(location => {
+        return location.city === newLocation.city && location.state === newLocation.state
+      })
+      if (foundLocation) {
+        const foundLocationIndex = this.locations.indexOf(foundLocation)
+        this.locations[foundLocationIndex] = newLocation
+      } else {
+        this.locations.push(newLocation)
+      }
+    },
+    retrieveLocalStorage () {
+      const saved = localStorage.getItem('savedLocations')
+      this.locations = JSON.parse(saved)
+      this.retrieveCurrent()
+    },
+    updateLocalStorage () {
+      const locationData = JSON.stringify(this.locations)
+      localStorage.setItem('savedLocations', locationData)
```

Comment:
This makes plenty of sense for our use of local storage right now - 
We've talked about an idea for future iterations in which we might want to have a separated view of stored locations, rather than persisting all on the main view unless manually deleted. Curious if this would provide an avenues for working around our limited API calls (seems to be ~ 4 calls / min allowed)